src/sonic-yang-models/yang-models/sonic-buffer-pg.yang

```diff
@@ -0,0 +1,66 @@
+module sonic-buffer-pg {
+	namespace "http://github.com/Azure/sonic-buffer-pg";
+	prefix bpg;
+
+	import sonic-extension {
+		prefix sonic-ext;
+	}
+
+	import sonic-port {
+		prefix prt;
+	}
+
+	import sonic-buffer-profile {
+		prefix bpf;
+	}
+
+	organization
+		"SONiC";
+
+	contact
+		"SONiC";
+
+	description
+		"SONIC BUFFER PG";
+
+	revision 2019-05-15 {
+		description
+			"Initial revision.";
+	}
+
+
+	container sonic-buffer-pg {
+
+		container BUFFER_PG {
+
+			list BUFFER_PG_LIST {
+				key "ifname pg_num";
+				//sonic-ext:key-pattern "BUFFER_PG|{ifname}|{pg_num}"; //special pattern used for extracting keys from
+				//redis-key and fill populate the yang instance
+				// Total list instance = number(key1) * number(key2) * number(key3)
+
+				leaf ifname {
+					type leafref {
+						path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:name";
+					}
+				}
+
+				leaf pg_num {
+					type string {
+						pattern "[0-7]((-)[0-7])?" {
```

Comment:
0-7

Reply:
What if there is an overlap PG entries i.e same PG number for more than one entry, is that a valid config? if no, how are we planning to validate that? thru custom validation?

Reply:
It should depend on the App code whether it needs to be overwritten or not. If we have 2 entries like "Ethernet4:3-4" and "Ethernet4:4-5", then both these configs are valid and should be applied in the order. Application in this case will be overriding  PG 4 with "Ethernet4:4-5" config. I am not aware if we can validate such overlap in yang.  The only solution I can think of is to split the PG(4 and 5) separately instead of range(say 4-5). This should be true not only for PG, but for other interfaces where we are specifying the range(Eg. Ethernet4-Ethernet10).